dawn/src/dawn/Compiler/DawnCompiler.cpp

```diff
@@ -310,6 +313,7 @@ DawnCompiler::optimize(std::map<std::string, std::shared_ptr<iir::StencilInstant
     // safe idea
     optimizer.pushBackPass<PassTemporaryType>();
     optimizer.pushBackPass<PassLocalVarType>();
+    optimizer.pushBackPass<PassRemoveScalars>();
```

Comment:
I think you are right, because when merging temporary fields we never decrease their scope. Therefore we never need to demote them to local variables. This also means that `PassTemporaryType` and `PassLocalVarType` are not needed, correct?

Reply:
agreed